explorer/testdata/for/for-loop-sum.carbon

```diff
@@ -0,0 +1,24 @@
+// Part of the Carbon Language project, under the Apache License v2.0 with LLVM
```

Comment:
Does this test anything interesting about `for` support that isn't covered by `for-loop.carbon`?